### PR TITLE
build: use cirrus for CI

### PR DESCRIPTION
**Pull request by JustinBeckwith**
#332 added Windows testing for CirrusCI, now let's move the rest of our CI from Circle over.

### CODE REVIEW

**Comment by fkorotkov** [comment]
For Windows stuff you can check #332. 😅

File: .cirrus.yml [hunk 1]
```diff
@@ -9,3 +11,37 @@ task:
...
+    - npm run codecov
+
+publish_task:
+  only_if: $CIRRUS_TAG =~ '^v.*'
+  depends_on:
+    - test_task
+    - lint_task
+    - windows_task
+  environment:
+    NPM_TOKEN: ENCRYPTED[188593d2219e49e28e05c983a768a7e89ef89441f07f450a58744161674b2e7f127caff0a8021bb5bb75a09d4e18750f]
```
Comment (ofrobots) [inline .cirrus.yml]: Does this have a race where both Circle and Cirrus would race to do a publish?

Comment (JustinBeckwith) [reply .cirrus.yml]: This PR deletes the CircleCI config... so I don't think so

File: .cirrus.yml [hunk 2]
```diff
@@ -9,3 +11,37 @@ task:
...
+
+coverage_task:
+  script:
+    - npm ci
+    - npm test
+    - npm run codecov
+
+publish_task:
+  only_if: $CIRRUS_TAG =~ '^v.*'
+  depends_on:
```
Comment (fkorotkov) [inline .cirrus.yml]: You don't need `_task` suffix. Please remove it. ;-)

Comment (fkorotkov) [reply .cirrus.yml]: Fixed in #345

**Pull request merged**